test/common/http/metadata_interface_test.cc

```diff
@@ -0,0 +1,51 @@
+#include <array>
+#include <cstdint>
+#include <string>
+
+#include "envoy/http/metadata_interface.h"
+
+#include "source/common/common/fmt.h"
+#include "source/common/common/logger.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using ::testing::HasSubstr;
+
+namespace Envoy {
+namespace Http {
+namespace {
+
+TEST(MetadataMapTest, KeyValueEscaped) {
```

Comment:
IIUC this test only validates the usage of "operator<<".
Can you add a test to validate that the formatter works properly? (maybe something that uses [`LogLevelSetter`](https://github.com/envoyproxy/envoy/blob/972b33265f26ed206c047fd05d9427f4cc6408aa/test/test_common/logging.h#L31) and [`EXPECT_LOG_CONTAINS`](https://github.com/envoyproxy/envoy/blob/main/test/test_common/logging.h#L163) will do the trick).

Reply:
it also validates that the ENVOY_LOG_MISC is not crashing, which, without this PR would crash.